autorest/date/timeRfc1123.go

```diff
@@ -0,0 +1,87 @@
+package date
+
+import (
+	"errors"
+	"time"
+)
+
+const (
+	rfc1123JSON = `"` + time.RFC1123 + `"`
+	rfc1123     = time.RFC1123
+)
+
+// Time defines a type similar to time.Time but assumes a layout of RFC1123 date-time (i.e.,
+// Mon, 02 Jan 2006 15:04:05 MST).
+
+type TimeRfc1123 struct {
```

Comment:
We are deriving types from ```time.Time ``` and adding/modifying some functions on those. If new types declare as named, they won't inherit ```time.Time ``` existing functions.

```Go
package main
import "fmt"
type newtype struct {
}
func (i newtype) print() {
	fmt.Println("newtype struct print")
}
type derivedtype1 newtype
type derivedtype2 struct {
	newtype
}
func main(){
	var t newtype
	t.print() // ---> newtype struct print
	
	var d2 derivedtype2
	d2.print() // ---> newtype struct print
	//var d1 derivedtype1  
	//d1.print() //// error: d1.print undefined (type derivedtype1 has no field or method print)
}